contracts/Staking.sol

```diff
@@ -403,13 +405,20 @@ contract Staking is ERCStaking, ERCStakingHistory, TimeHelpers, IsContract {
     }
 
     /*
-    function multicall(bytes[] _calls) public {
+     * From here: https://etherscan.io/address/0xa86ba3b6d83139a49b649c05dbb69e0726db69cf#code
+     * TODO: UnimplementedFeatureError: Nested arrays not yet implemented.
+    function multicall(bytes[] memory_calls) public {
```

Comment:
Following up what I mentioned above, that [bug](https://blog.ethereum.org/2019/03/26/solidity-optimizer-and-abiencoderv2-bug/) was actually related to `bytesNN` types handling. Although, it may not affect us, since it seems to be related only to storage references, it tells you how critical it could have been.

At least, I would migrate to solc `0.5.x` to make sure we are using the latests `ABIEncoderV2` bug-fixes.